### PR TITLE
Start moving gatsby-telemetry to typescript

### PR DESCRIPTION
**Pull request by jamo**
This tries to be the minimal set of changes to get this to typescript and make writing new parts be possible in TS

### CODE REVIEW

File: packages/gatsby-telemetry/src/telemetry.ts [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+  min: number
+  max: number
+  sum: number
+  mean: number
+  median: number
+  stdDev: number
+  skewness: number
+}
+
+module.exports = class AnalyticsTracker {
```
Comment (blainekasten) [inline packages/gatsby-telemetry/src/telemetry.ts:42]: can we change this to a named export?

File: packages/gatsby-telemetry/src/telemetry.ts [hunk 2]
```diff
@@ -0,0 +1,353 @@
+const uuidv4 = require(`uuid/v4`)
+const EventStorage = require(`./event-storage`)
+const { cleanPaths } = require(`./error-helpers`)
+const { isCI, getCIName } = require(`gatsby-core-utils`)
+const os = require(`os`)
+const { join, sep } = require(`path`)
+const isDocker = require(`is-docker`)
+const showAnalyticsNotification = require(`./showAnalyticsNotification`)
+const lodash = require(`lodash`)
```
Comment (blainekasten) [inline packages/gatsby-telemetry/src/telemetry.ts:9]: can we change these all to module imports? typescript can't follow types from modules with require. You have to use `import` to get their types

File: packages/gatsby-telemetry/src/telemetry.js [hunk 3]
```diff
@@ -312,12 +274,10 @@ module.exports = class AnalyticsTracker {
...
-
   async sendEvents() {
     if (!this.isTrackingEnabled()) {
-      return Promise.resolve()
+      return Promise.resolve(true)
     }
-
     return this.store.sendEvents()
   }
 }
```
Comment (blainekasten) [inline packages/gatsby-telemetry/src/telemetry.js:283]: shouldn't this file be deleted?? why do we have a .js and a .ts?

**Comment by jamo** [comment]
Thanks @blainekasten, all good points and ideas. I'll tackle these next week.

File: packages/gatsby-telemetry/src/telemetry.js (hunk 3 above)
Comment (jamo) [reply packages/gatsby-telemetry/src/telemetry.js]: Done

File: packages/gatsby-telemetry/src/telemetry.ts (hunk 2 above)
Comment (jamo) [reply packages/gatsby-telemetry/src/telemetry.ts]: Converted most of them. There is some more work to be done to fully type this but I wanted to start by getting this over to TS to make future development easier

**Pull request merged**